packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1741,14 +1741,26 @@ class AnnotatedRegionLayer<T> extends ContainerLayer {
   ///
   /// If not provided, all offsets are considered to be contained within this
   /// layer, unless an ancestor layer applies a clip.
+  ///
+  /// If [offset] is set, then the offset is applied to the size region before
+  /// hit testing in [find].
   final Size size;
 
+  /// The [offset] is optionally used to translate the clip region for the
+  /// hit-testing of [find] by [offset].
+  ///
+  /// If not provided, all offsets are considered to be contained within this
+  /// layer, unless an ancestor layer applies a clip.
+  ///
+  /// Ignored if [size] is not set.
+  final Offset offset;
+
   @override
   S find<S>(Offset regionOffset) {
     final S result = super.find<S>(regionOffset);
     if (result != null)
       return result;
-    if (size != null && !size.contains(regionOffset))
+    if (size != null && !((offset ?? Offset.zero) & size).contains(regionOffset))
```

Comment:
rather than defaulting it here, default it in the constructor and assert non-null. it'll be more efficient.

Reply:
Good point, thanks.